test/test.c

```diff
@@ -20243,6 +20248,15 @@ test_ForcedReconnect(void)
     natsMsg *msg = NULL;
     natsPid pid = NATS_INVALID_PID;
 
+#ifdef __SANITIZE_THREAD__
+    // threadSanitizer complains that we close the fd when a read may be in
+    // progress. Since it appears to work as desired, skipping the test.
```

Comment:
Well this is concerning in that we are testing a public function (natsConnection_Reconnect()). If this is causing a sanitize thread error, then maybe the approach of closing the socket is wrong? I mean it will cause the socket read error for sure, which triggers the reconnect so it works well, but it is technically wrong I guess? Would the sanitize complain (and would the feature still work) if we had use `natsSock_Shutdown()` instead of `natsSock_Close()`?

Reply:
So I tried with shutdown instead of close and it seems to have solve the warning. Would have to test on all platforms to see if that introduces an issue or not.

Reply:
I'll make the change once you confirm, probably as a separate PR that fill fix the issue and re-enable the test. It will keep this one test-only, for release notes and such?

Reply:
Sure, can be done in separate PR. I did not mean that I was going to check on several platforms myself. I guess the run on GA that runs on Windows should verify that it works. But I think that we don't have Windows yet, so let me try on my Windows VM and verify.

Reply:
Ok, it works and by the way, in that other PR we can also alter the test a little bit by setting the `natsOptions_SetReconnectWait(opts, 100)` to speed-up the test. Otherwise it waits for the default reconnect wait interval (2sec) since it reconnects to the same server.

Reply:
Because the ifdef-skip didn't work on Travis as intended, I added the Reconnect changes here (rather than figuring out how to skip properly, since we'd be un-skipping anyway).